c-dependencies/js-compute-runtime/builtins/crypto-algorithm.cpp

```diff
@@ -10,6 +11,83 @@ namespace builtins {
 
 namespace {
 
+const EVP_MD *createDigestAlgorithm(JSContext *cx, JS::HandleObject key) {
+
+  JS::RootedObject alg(cx, CryptoKey::get_algorithm(key));
+
+  JS::RootedValue hash_val(cx);
+  JS_GetProperty(cx, alg, "hash", &hash_val);
+  JS::RootedObject hash(cx, &hash_val.toObject());
+  JS::RootedValue name_val(cx);
+  JS_GetProperty(cx, hash, "name", &name_val);
```

Comment:
In our current implementation yes but that is only because we only support CryptoKeys which require a hash algorithm.

To future proof this I have added defensive code which will report an error if there is no hash algorithm 👍